dataset/src/main/scala/frameless/functions/NonAggregateFunctions.scala

```diff
@@ -279,4 +278,13 @@ trait NonAggregateFunctions {
   def upper[T](str: TypedColumn[T, String]): TypedColumn[T, String] = {
     new TypedColumn[T, String](untyped.upper(str.untyped))
   }
-}
+
+  /**
+    * Non-Aggregate function: Extracts the year as an integer from a given date/timestamp/string.
+    *
+    * apache/spark
+    */
+  def year[T](col: TypedColumn[T, String]): TypedColumn[T, Int] = {
```

Comment:
This should probably `TypedColumn[T, Option[Int]]`

Reply:
I agree here. This should fail when there is years to be extracted. Is Spark returns null in this case, then this should be encoded as  @frosforever suggests. Should be a trivial change.